src/plugins/intel_cpu/src/emitters/plugin/x64/utils.hpp

```diff
@@ -9,84 +9,35 @@
 namespace ov {
 namespace intel_cpu {
 
-// Usage
-// 1. Include this headfile where JIT kennels of CPU plugin are implemented for Register printing
-// 2. Invoke RegPrinter::print method. Here are some examples. Note that user friendly register name
-//    will be printed, if it has been set. Current implementation doesn't buffer the name. So if you
-//    choose to set a name for the register, do not use local variable to pass the name, just pass a
-//    direct string to the interface like examples. While Original Xbyak register name will always be
-//    printed.
-//    Example 1:
-//    Invocation: RegPrinter::print<float>(*this, vmm_val, "vmm_val");
-//    Console:    vmm_val | ymm0: {30, 20, 25, 29, 24, 31, 27, 23}
-//
-//    Example 2:
-//    Invocation: RegPrinter::print<float>(*this, vmm_val);
-//    Console:    ymm0: {30, 20, 25, 29, 24, 31, 27, 23}
-//
-//    Example 3:
-//    Invocation: RegPrinter::print<int>(*this, vmm_idx, "vmm_idx");
-//    Console:    vmm_idx | ymm1: {5, 6, 0, 2, 0, 6, 6, 6}
-//
-//    Example 4:
-//    Invocation: RegPrinter::print<int>(*this, reg_work_amount, "reg_work_amount");
-//    Console:    reg_work_amount | r13: 8
-//
-//    Example 5:
-//    Invocation: RegPrinter::print<int>(*this, reg_work_amount);
-//    Console:    r13: 8
-//
-//    Example 6:
-//    Invocation: RegPrinter::print<float>(*this, reg_tmp_64, "reg_tmp_64");
-//    Console:    reg_tmp_64 | r15: 1
-//
-// Parameter
-// The following combinations of Register types and precisions are supported.
-//          fp32        int32       int8        u8
-// Xmm      Yes         Yes         No          No
-// Ymm      Yes         Yes         No          No
-// Zmm      Yes         Yes         No          No
-// Reg64    Yes         Yes         No          No
-// Reg32    Yes         Yes         No          No
-// Reg16    No          No          Yes         Yes
-// Reg8     No          No          Yes         Yes
-
-class RegPrinter {
+// The class emit register spills for the possible call of external binary code
+class EmitABIRegSpills {
 public:
-    using jit_generator = dnnl::impl::cpu::x64::jit_generator;
-    template <typename PRC_T, typename REG_T,
-    typename std::enable_if<std::is_base_of<Xbyak::Xmm, REG_T>::value, int>::type = 0>
-    static void print(jit_generator &h, REG_T reg, const char *name = nullptr) {
-        print_vmm<PRC_T, REG_T>(h, reg, name);
-    }
-    template <typename PRC_T, typename REG_T,
-    typename std::enable_if<!std::is_base_of<Xbyak::Xmm, REG_T>::value, int>::type = 0>
-    static void print(jit_generator &h, REG_T reg, const char *name = nullptr) {
-        print_reg<PRC_T, REG_T>(h, reg, name);
-    }
+    EmitABIRegSpills(dnnl::impl::cpu::x64::jit_generator* h);
```

Comment:
if I am not mistaken, it was agreed to introduce boolean state whose value will be changed during preamble/postamble, and whose value will be checked in class destructor (not sure but probably it is worth to create second state for rsp_align/restore tracking). Please correct me if I am wrong

Reply:
Oh, thank you for the reminder! You're right!
Added, thank you one more time!